rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -708,8 +708,25 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 	const CGameHelper::BuildSquareStatus tbs = CGameHelper::TestUnitBuildSquare(buildInfo, feature, -1, true);
 
 	switch (tbs) {
-		case CGameHelper::BUILDSQUARE_OPEN:
-			break;
+		case CGameHelper::BUILDSQUARE_OPEN: {
+			// <pos> might map to a non-blocking portion
+			// of the buildee's yardmap, fallback check
+			const CUnit* u = CGameHelper::GetClosestFriendlyUnit(nullptr, buildInfo.pos, 8.0f, allyteam);
```

Comment:
I see yardmap referred to in the comment above, is the magic  `8.0f` actually `SQUARE_SIZE`?
```suggestion
			const CUnit* u = CGameHelper::GetClosestFriendlyUnit(nullptr, buildInfo.pos, static_cast<float>(SQUARE_SIZE), allyteam);
```

Reply:
No but it can be.